src/blockdata/transaction.rs

```diff
@@ -609,6 +616,60 @@ impl Transaction {
     pub fn is_explicitly_rbf(&self) -> bool {
         self.input.iter().any(|input| input.sequence < (0xffffffff - 1))
     }
+
+    /// Returns the nLockTime as an absolute timelock.
+    pub fn absolute_lock_time(&self) -> timelock::Abs {
+        timelock::Abs::from(self.lock_time)
+    }
+
+    /// Returns `true` if this transactions nLockTime should be ignored.
+    pub fn ignore_lock_time(&self) -> bool {
+        if self.lock_time == 0 {
+            return true;
+        }
+        self.input.iter().all(|input| input.sequence == 0xffffffff)
+    }
+
+    /// Returns the most strict relative timelocks that must be satisfied for this transaction.
+    ///
+    /// # Returns
+    ///
+    /// `(t1, t2)`, where t1 is the most strict block based relative timelock and t2 is the most
+    /// strict time based relative timelock.
+    ///
+    /// `None` if timelock was not found, `Some(t)` if a timelock was found (including zero value
+    /// timelocks).
+    pub fn maximum_relative_lock_time(&self) -> (Option<timelock::Rel>, Option<timelock::Rel>) {
+        let mut max_block = None;
+        let mut max_time = None;
+
+        for input in self.input.iter() {
```

Comment:
This could probably be more like:

```rust
  use timelock::Relative::*;
  input.iter().fold((None, None), |(block, time), input| {
   match t.relative_timelock() {
      Some(t @ Time { .. }) => (block, time.or_else(t).min(t).expect("same type")),
      Some(t @ Block { .. }) => (block.or_else(t).min(t).expect("same type"), time),
   }
}
```

+/- `Relative::min`